include/af/random_engine.h

```diff
@@ -0,0 +1,327 @@
+       \param[in]   ndims The number of dimensions read from the \p dims parameter
+       \param[in]   dims A C pointer with \p ndims elements. Each value represents the size of that dimension
+       \param[in]   type The type of the \ref af_array object
+
+       \returns \ref AF_SUCCESS if the execution completes properly
+    */
+    AFAPI af_err af_random_engine_normal(af_array *out, af_random_engine engine, const unsigned ndims, const dim_t * const dims, const af_dtype type);
+
+    /**
+       C Interface for setting the seed of a random engine
+
+       \param[out]  engine is the random engine object
+       \param[in]   seed is the initializing seed of the random number generator
+
+       \returns \ref AF_SUCCESS if the execution completes properly
+    */
+    AFAPI af_err af_random_engine_set_seed(af_random_engine *engine, const uintl seed);
+
+    /**
+       C Interface for setting the type of the default random engine
+
+       \param[in]   rtype is the type of the random number generator
+
+       \returns \ref AF_SUCCESS if the execution completes properly
+    */
+    AFAPI af_err af_default_random_engine_set_type(const af_random_type rtype);
+
+    /**
+       C Interface for getting the seed of a random engine
+
+       \param[out]  out The pointer to the returned seed.
+       \param[in]   engine is the random engine object
+
+       \returns \ref AF_SUCCESS if the execution completes properly
+    */
+    AFAPI af_err af_random_engine_get_seed(uintl * const seed, af_random_engine engine);
+
+    /**
+       C Interface for releasing random engine
+
+       \param[in] engine is the random engine object
+       \returns \ref AF_SUCCESS if the execution completes properly
+    */
+    AFAPI af_err af_release_random_engine(af_random_engine engine);
+
+    //General rand calls
+
+    /**
+        \param[out] out is the generated array
+        \param[in] ndims is size of dimension array \p dims
+        \param[in] dims is the array containing sizes of the dimension
+        \param[in] type is the type of array to generate
+
+       \ingroup data_func_randu
+    */
+    AFAPI af_err af_randu(af_array *out, const unsigned ndims, const dim_t * const dims, const af_dtype type);
+
+    /**
+        \param[out] out is the generated array
+        \param[in] ndims is size of dimension array \p dims
+        \param[in] dims is the array containing sizes of the dimension
+        \param[in] type is the type of array to generate
+
+       \ingroup data_func_randn
+    */
+    AFAPI af_err af_randn(af_array *out, const unsigned ndims, const dim_t * const dims, const af_dtype type);
+
+    /**
+        \param[in] seed is a 64 bit unsigned integer
+
+        \ingroup data_func_setseed
+    */
+    AFAPI af_err af_set_seed(const uintl seed);
+
+    /**
+        \param[out] seed which is a 64 bit unsigned integer
+
+        \ingroup data_func_getseed
+    */
+    AFAPI af_err af_get_seed(uintl *seed);
+
+#ifdef __cplusplus
+}
+#endif
```

Comment:
Should be move all these functions being in `data_func` to `random_func` group in documentation?